### PR TITLE
Add Contributing Guidelines

### PR DESCRIPTION
**Pull request by Snugug**
Adds contributing guidelines to the project. Heavily inspired by Atom's contributing guidelines. We can have a discussion about content, but I think it's important we get these in before 2.0

### CODE REVIEW

**Comment by Snugug** [comment]
Closes last task of #247

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+* Include relevant Emoji from our [Emoji cheatsheet](#emoji-cheatsheet)
+
+### Branching Model
+
+* Branches must be made off of the most current `master` branch from `git@github.com:dlmanning/gulp-sass.git`
+* Pull requests must be made into our [master](https://github.com/dlmanning/gulp-sass/tree/master) branch.
+* The following branch prefixes should be used when creating a new branch:
+  * `hotfix/` - fixes a bug that got through to a live version and need to be squashed
+  * `feature/` - update to or new code for `gulp-sass`
+  * `release/` - a release of `gulp-sass`
```
Comment (Keats) [inline CONTRIBUTING.md]: Do we really care about that?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+* Do not squash or rebase your commits when submitting a Pull Request. It makes it much harder to follow your work and make incremental changes.
+* Update the [CHANGELOG](#maintaining-thechangelog) with your changes.
+* Ensure no Emoji tags are used in the title of your Pull Request
+
+### Git Commit Messages
+
+* Use the present tense (`"Add feature"` not `"Added Feature"`)
+* Use the imperative mood (`"Move cursor to…"` not `"Moves cursor to…"`)
+* Limit the first line to 72 characters or less
+* Include relevant Emoji from our [Emoji cheatsheet](#emoji-cheatsheet)
```
Comment (Keats) [inline CONTRIBUTING.md]: I always found this a bit silly on javascript repos, most of the time I'm looking at the log in my terminal, not on github

**Comment by Keats** [comment]
Rest looks good otherwise

File: CONTRIBUTING.md (hunk 1 above)
Comment (dlmanning) [inline CONTRIBUTING.md]: Yeah, this seems like overkill for a project this size

Comment (Snugug) [inline CONTRIBUTING.md]: I'll reduce this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Snugug) [inline CONTRIBUTING.md]: Which part are you talking about? The commit message, the PR stuff, or what?

The PR stuff I care a lot about. It provides order to contributing which is insanely important when trying to manage a project of any size, regardless of the language it's written in.

Comment (Snugug) [inline CONTRIBUTING.md]: I like Emoji. I spend most of my project management time on GitHub, but either way, it's an easy way to get a quick idea of what's happening in a commit. I'll reduce it's importance

**Comment by Snugug** [comment]
Removed branching model stuff. Recheck?

**Comment by dlmanning** [comment]
lgtm

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+* Use the present tense (`"Add feature"` not `"Added Feature"`)
+* Use the imperative mood (`"Move cursor to…"` not `"Moves cursor to…"`)
+* Limit the first line to 72 characters or less
+* Consider including relevant Emoji from our [Emoji cheatsheet](#emoji-cheatsheet)
+
+## Creating a New Version
+
+Versioning is done through [SEMVER](http://semver.org/). When creating a new version, create new release branch off of `master` with the version's name, and create a new tag with `v` prefixed with the version's name from that branch. 
+
+For instance, if you are creating version `1.1.0`, you would start by merging `develop` into `master`, create a branch `release/1.1.0` from `master`, and create a tag `v1.1.0` from branch `release/1.1.0`.
```
Comment (Keats) [inline CONTRIBUTING.md]: we don't have a develop branch

Comment (Snugug) [inline CONTRIBUTING.md]: Fair point

**Comment by Snugug** [comment]
Updated

**Comment by Snugug** [comment]
If we can get this merged in, we can release 2.0

**Pull request merged**